rules/no-fixed.js

```diff
@@ -17,28 +17,22 @@ module.exports = {
 	},
 
 	create: function(context) {
-		function inspectDeclarationExpression(emitted) {
+		function inspectType(emitted) {
 			if (emitted.exit) {
 				return;
 			}
 
-			let node = emitted.node;
-			let variableType = node.literal.literal;
-
-			if (typeof variableType !== "string") {
-				return;
-			}
-
-			if (variableType.indexOf("fixed") === 0 || variableType.indexOf("ufixed") === 0) {
+			let type = emitted.node.literal;
+			if (type === "fixed" || type === "ufixed") {
```

Comment:
http://solidity.readthedocs.io/en/develop/types.html#fixed-point-numbers
Fixed point numbers can have a MxN pattern, so even `ufixed128x19` is valid but will not be caught. Please replace the exact match by a prefix match (like the `indexOf` match that currently exists).

`if (type.indexOf("fixed") === 0 || type.indexOf("ufixed") === 0)`

I should've added tests for this, will do that once I merge your PR.

Reply:
Good catch! Fixed that up